service/src/main/java/eu/europa/ec/fisheries/ers/service/facatch/evolution/DepCatchProgressHandler.java

```diff
@@ -0,0 +1,21 @@
+package eu.europa.ec.fisheries.ers.service.facatch.evolution;
+
+import eu.europa.ec.fisheries.ers.fa.entities.FaCatchEntity;
+import eu.europa.ec.fisheries.ers.fa.entities.FishingActivityEntity;
+import eu.europa.ec.fisheries.ers.fa.utils.FaReportDocumentType;
+import eu.europa.ec.fisheries.ers.service.dto.fishingtrip.catchprogress.CatchProgressDTO;
+import eu.europa.ec.fisheries.ers.service.dto.fishingtrip.catchprogress.DepCatchProgressDTO;
+
+public class DepCatchProgressHandler extends CatchProgressHandler {
+    @Override
+    public CatchProgressDTO prepareCatchProgressDTO(FishingActivityEntity fishingActivity) {
+        String reportType = fishingActivity.getFaReportDocument().getTypeCode();
+        DepCatchProgressDTO catchProgressDTO = new DepCatchProgressDTO(fishingActivity.getFaReportDocument().getTypeCode(), reportType.equals(FaReportDocumentType.DECLARATION.name()), fishingActivity.getCalculatedStartTime());
+
+        for (FaCatchEntity faCatch : fishingActivity.getFaCatchs()) {
+            handleSimpleCatch(faCatch, catchProgressDTO);
+        }
+
+        return catchProgressDTO;
+    }
+}
```

Comment:
Finally I see that a lot of classes ending in something**CatchProgressHandler** share the same code . I would suggest to move similar code to superclass with generics